### PR TITLE
Remove GCR from everywhere.

### LINKED ISSUE #1434: Migrate :master image building from jenkins to github actions
Currently our `:master` and tagged (see [./docs/tagging.md]) builds are built using an internal jenkins instance that is going away soon. We need to move the service that builds our `:master` and release builds elsewhere. Additionally, our gcr.io image repository contains heptio branding so we need to move away from that.

TODO:
- [x] register `projectcontour` on the docker hub, figure out how to distribute write privs to the team
- [x] set up a `on commit to master` build that builds `docker.io/projectcontour/contour:master`
- [x] figure out how to build from a tag and tag an image as `:latest`
- [ ] grep through all the documentation and update all the `gcr.io` images
- [x] update release documentation (on internal wiki -- ask @davecheney) - BONUS POINTS will be awarded if the release documentation is moved to this repo.

Out of scope:

- [ ] Do not disable jenkins jobs, they can continue to push to gcr.io, 0.15 will be the last image we publish there.

### PR DESCRIPTION
**Pull request by youngnick**
Fixes #1434

Signed-off-by: Nick Young <ynick@vmware.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 PROJECT = contour
 MODULE = github.com/heptio/$(PROJECT)
-REGISTRY ?= gcr.io/heptio-images
+REGISTRY ?= projectcontour
```
Comment (davecheney) [inline Makefile]: is it worth being concrete here; `docker.io/projectcontour` ?

File: docs/tagging.md [hunk 2]
```diff
@@ -4,13 +4,13 @@ This document describes Contour's image tagging policy.
 
 ## Released versions
 
-`gcr.io/heptio-images/contour:<SemVer>`
+`projectcontour/contour:<SemVer>`
 
-Contour follows the [Semantic Versioning](http://semver.org/) standard for releases. Each tag in the github.com/heptio/contour repository has a matching image. eg. `gcr.io/heptio-images/contour:v0.1.0`
+Contour follows the [Semantic Versioning](http://semver.org/) standard for releases. Each tag in the github.com/heptio/contour repository has a matching image. eg. `projectcontour/contour:v0.1.0`
```
Comment (davecheney) [inline docs/tagging.md]: same comment as above, should we be absolute `docker.io/projectcontour/contour:v0.15.0` ?

Also, use 0.15.0

File: docs/tagging.md [hunk 3]
```diff
@@ -4,13 +4,13 @@ This document describes Contour's image tagging policy.
...
 
 ### Latest
 
-`gcr.io/heptio-images/contour:latest`
+`projectcontour/contour:latest`
```
Comment (davecheney) [inline docs/tagging.md]: same as above

File: docs/tagging.md [hunk 4]
```diff
@@ -4,13 +4,13 @@ This document describes Contour's image tagging policy.
...
 
 The `latest` tag follows the most recently released version of Contour.
```
Comment (davecheney) [inline docs/tagging.md]: most recent stable version of Contour.

File: docs/upgrading.md [hunk 5]
```diff
@@ -50,7 +50,7 @@ This section contains information for administrators who wish to apply the Conto
 
 ### Upgrade to Contour 0.15.0
 
-Change the Contour image version to `gcr.io/heptio-images/contour:v0.15.0`.
+Change the Contour image version to `projectcontour/contour:v0.15.0`.
```
Comment (davecheney) [inline docs/upgrading.md]: Please expand on this. perhaps

Due to the sun setting on the Heptio brand from v0.15.0 onwards our images are now served from the docker hub (link).

File: Makefile (hunk 1 above)
Comment (youngnick) [reply Makefile]: it won't hurt anything, I guess. I don't see the assumption that 'no domain == docker hub' changing now though.

**Comment by davecheney** [comment]
please use your best judgement if docker.io/projectcontour or the bare projectcontour is appropriate. all I ask is that it is consistent across everywhere we write it

**Pull request merged**